### PR TITLE
Handle replicate workflow state with exist workflow data

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Handle replicate workflow state with exist workflow data.

**Why?**
During migration interruption, some workflow data might be partially migrated. This case is not handle in replicate workflow state.

**How did you test it?**
Update current unit test. 

A manual test will be perform: 
1. Connect two cluster.
2. Start a long running workflow.
3. Disconnect two cluster.
4. Terminate the old workflow and start a new long running workflow.
5. Force replication open workflow.
6. Force replication closed workflow.

**Potential risks**

**Is hotfix candidate?**
Yes.

### CODE REVIEW

File: service/history/ndc/history_replicator.go [hunk 1]
```diff
@@ -303,6 +303,35 @@ func (r *HistoryReplicatorImpl) ApplyWorkflowState(
...
+	case nil:
+		// workflow exists, do resend.
+		currentVersionHistory, err := versionhistory.GetCurrentVersionHistory(ms.GetExecutionInfo().GetVersionHistories())
+		if err != nil {
+			return err
+		}
+		lastVersionHistory, err := versionhistory.GetLastVersionHistoryItem(currentVersionHistory)
+		if err != nil {
+			return err
+		}
```
Comment (yux0) [inline service/history/ndc/history_replicator.go:320]: Check if the local last version history and remote last version history are identical

**Review by wxing1292 — APPROVED** [review]
plz address the comments about API idempotency then land the PR

File: tests/ndc/ndc_integration_test.go [hunk 2]
```diff
@@ -402,6 +403,80 @@ func (s *nDCIntegrationTestSuite) TestEmptyVersionAndNonEmptyVersion() {
 	)
 }
 
+func (s *nDCIntegrationTestSuite) TestReplicateWorkflowState_PartialReplicated() {
+
+	s.setupRemoteFrontendClients()
+	workflowID := "ndc-single-branch-test" + uuid.New()
```
Comment (wxing1292) [inline tests/ndc/ndc_integration_test.go:409]: plz make the workflow ID unique, e.g. by naming it `replicate-workflow-state-partially-replicated`

**Review by wxing1292 — APPROVED** [review]
LGTM

File: service/history/replication/task_executor.go
Comment (yycptt) [inline service/history/replication/task_executor.go:1]: is this a different code path than executable_workflow_state_task_test.go for replication task execution? Is there a plan for consolidating them?

**Review by yycptt — APPROVED** [review]

File: service/history/replication/task_executor.go
Comment (yux0) [reply service/history/replication/task_executor.go:1]: This file will be deprecated once we move to streaming replication.

**Pull request merged**